core/src/main/java/io/undertow/Undertow.java

```diff
@@ -250,14 +253,15 @@ public XnioWorker getWorker() {
         AJP
     }
 
-    private static class ListenerConfig {
+    public static class ListenerConfig {
```

Comment:
I think it would be better to add an Undertow.ListenerBuilder class, and just expose that as the public API, then have addListener(ListenerBuilder). I just don't really like the idea of a user having to deal with 3 levels of nestled classes (new Undertow.ListenerConfig.Builder())